### ISSUE TITLE
1.19.1 - weird linking issues

### ISSUE THREAD

**Issue by croth1** [conda/conda-build#823]
The new conda-build version introduces weird linking bugs on OS X.

Here is a diff of the dynamically linked libraries between versions 1.19.0 and 1.19.1. This is the recipe I am working on: https://github.com/bioconda/bioconda-recipes/pull/1021

```diff
diff --git a/tmp/gnuplot_otool b/tmp/gnuplot_otool_new
index fa0996e..f2ce21f 100644
--- a/tmp/gnuplot_otool
+++ b/tmp/gnuplot_otool_new
@@ -1,13 +1,13 @@
-../bin/gnuplot:
+_test/bin/gnuplot:
        /usr/lib/libedit.3.dylib (compatibility version 2.0.0, current version 3.0.0)
        @rpath/./libncurses.5.dylib (compatibility version 5.0.0, current version 5.0.0)
        @rpath/./libz.1.dylib (compatibility version 1.0.0, current version 1.2.8)
        @rpath/./libgd.3.dylib (compatibility version 4.0.0, current version 4.0.0)
        @rpath/./libjpeg.8.dylib (compatibility version 13.0.0, current version 13.0.0)
        @rpath/./libpng16.16.dylib (compatibility version 34.0.0, current version 34.0.0)
-       @rpath/libfreetype.6.dylib (compatibility version 18.0.0, current version 18.4.0)
-       @rpath/libtiff.5.dylib (compatibility version 8.0.0, current version 8.4.0)
+       libfreetype.6.dylib (compatibility version 18.0.0, current version 18.4.0)
+       libtiff.5.dylib (compatibility version 8.0.0, current version 8.4.0)
        /usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
        /usr/local/opt/lua/lib/liblua.5.2.dylib (compatibility version 5.2.0, current version 5.2.4)
        /usr/lib/libstdc++.6.dylib (compatibility version 7.0.0, current version 104.1.0)
-       /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
+       @rpath/libfreetype.6.dylib (compatibility version 1.0.0, current version 1226.10.1)
```

**Comment by stuarteberg** [comment]
Probably related to #818.

**Comment by croth1** [comment]
This is after setting `rpath = ' -Wl,-rpath,%(PREFIX)s/lib' % d`

**Comment by stuarteberg** [comment]
Hmm... well, presumably #808 is the place to start looking.  I guess you could try reverting that changeset in a local copy of conda-build, and see if that fixes the issue.  If so, you'd have a good argument that #808 needs to be reverted for now.

**Comment by croth1** [comment]
Bisecting: 0 revisions left to test after this (roughly 0 steps)
[0c0dd6edca285e21ac619c46d3f10fec85126855] macho.py: re-work otools

**Comment by stuarteberg** [comment]
OK, so @mingwandroid, do you have any comments on this?

**Comment by mingwandroid** [comment]
Can you try, or are you trying with git master? I put a fix down earlier for a pretty dumb mistake.

**Comment by croth1** [comment]
@mingwandroid, sorry I assumed that master was bad. That's actually not the case.

394e6c125478a9c3436516552bb9e6ee62744778 seems to fix this.

**Comment by groutr** [comment]
Closing this issue as it seems it has already been resolved.

**Comment by croth1** [comment]
@groutr can we have a bugfix release soon? I am blocked until the new release is out.

**Comment by kalefranz** [comment]
Need to fix #807 and hopefully have a release out today.

**Comment by stuarteberg** [comment]
Definitely merge #824 before the release, otherwise Windows users are hosed.

(Not sure why Travis is now refusing to even *start* those tests, but they were passing before I made my latest change, which was very minor.  That PR is good to go.)
